docs/architecture/grpc-for-wcf-developers/apm.md

```diff
@@ -1,11 +1,221 @@
+
+## Logging vs Metrics
+
+Before looking at the implementation details, it is necessary to understand the difference between logging and metrics.
+
+Logging is concerned with text messages that record detailed information about things that have happened in the system. Log messages may include exception data like stack traces, or structured data that provides context about the message. Logging output is commonly written to a searchable text store.
+
+Metrics refers to numeric data that is designed to be aggregated and presented using charts and graphs in a dashboard that provides a view of the overall health and performance of an application. Metrics data can also be used to trigger automated alerts when a threshold is exceeded. Examples of metrics data include the time taken to process requests, the number of requests per second being handled by an instance of a service, or the number of failed requests on an instance.
+
+## Logging in ASP.NET Core gRPC
+
+ASP.NET Core provides support for logging out-of-the-box, in the form of the [Microsoft.Extensions.Logging package](https://www.nuget.org/packages/Microsoft.Extensions.Logging). This is included with the Web SDK so there is no need to install it manually. The gRPC framework writes its internal diagnostic logging messages to ASP.NET Core logging so they can be processed/stored along with the application's own messages.
+
+### Producing log messages
+
+The Logging extension is automatically registered with ASP.NET Core's dependency injection system, so loggers can be specified as a constructor parameter on gRPC service types.
+
+```csharp
+public class StockData : Stocks.StocksBase
+{
+    private readonly ILogger<StockData> _logger;
+
+    public StockData(ILogger<StockData> logger)
+    {
+        _logger = logger;
+    }
+}
+```
+
+A lot of log messages around requests, exceptions, etc. are provided by the ASP.NET Core and gRPC framework components. You should add your own log messages to provide detail and context about application logic rather than lower level concerns.
+
+For more information please refer to the [ASP.NET Core Logging documentation](https://docs.microsoft.com/aspnet/core/fundamentals/logging/?view=aspnetcore-3.0).
+
+## Metrics in ASP.NET Core gRPC
+
+The .NET Core framework provides a set of components for measuring performance in the [System.Diagnostics.Tracing](https://www.nuget.org/packages/System.Diagnostics.Tracing/) package. This includes the `EventSource` and `EventCounter` types, which can be used to emit basic numeric data that can be consumed by external processes like the [dotnet-counters global tool](https://github.com/dotnet/diagnostics/blob/master/documentation/dotnet-counters-instructions.md), or Event Tracing for Windows. To learn more about using `EventCounter` in your own code refer to the [EventCounter Introduction Tutorial](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Tracing/documentation/EventCounterTutorial.md).
```

Comment:
The types are available as part of the BCL. I'd omit mentioning the package altogether
```suggestion
The .NET Core runtime provides a set of components for emitting and observing metrics. This includes the `EventSource` and `EventCounter` types, which can be used to emit basic numeric data that can be consumed by external processes like the [dotnet-counters global tool](https://github.com/dotnet/diagnostics/blob/master/documentation/dotnet-counters-instructions.md), or Event Tracing for Windows. To learn more about using `EventCounter` in your own code refer to the [EventCounter Introduction Tutorial](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Tracing/documentation/EventCounterTutorial.md).
```